src/rgw/rgw_process.cc

```diff
@@ -305,6 +306,11 @@ int process_request(rgw::sal::RGWRadosStore* const store,
     *http_ret = s->err.http_ret;
   }
   int op_ret = 0;
+
+  if(user != nullptr) {
+    *user = s->user->get_id().to_str();
```

Comment:
`s->user` is also unchecked, and i'm not sure we guarantee that it gets initialized for all requests. maybe something like this?
```c++
if (user && rgw::sal::RGWObject::empty(s->user)) {
  *user = s->user->get_id().to_str();
}
```